Obsolete rollup-plugin-sourcemap

When running `npm i` there is a bunch of warnings:

```
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @rollup/pluginutils@3.1.0
npm warn Found: rollup@4.22.1
npm warn node_modules/rollup
npm warn   dev rollup@"^4.22.1" from the root project
npm warn   9 more (@rollup/plugin-commonjs, @rollup/plugin-json, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer rollup@"^1.20.0||^2.0.0" from @rollup/pluginutils@3.1.0
npm warn node_modules/rollup-plugin-sourcemaps/node_modules/@rollup/pluginutils
npm warn   @rollup/pluginutils@"^3.0.9" from rollup-plugin-sourcemaps@0.6.3
npm warn   node_modules/rollup-plugin-sourcemaps
npm warn
npm warn Conflicting peer dependency: rollup@2.79.1
npm warn node_modules/rollup
npm warn   peer rollup@"^1.20.0||^2.0.0" from @rollup/pluginutils@3.1.0
npm warn   node_modules/rollup-plugin-sourcemaps/node_modules/@rollup/pluginutils
npm warn     @rollup/pluginutils@"^3.0.9" from rollup-plugin-sourcemaps@0.6.3
npm warn     node_modules/rollup-plugin-sourcemaps
```

This happens because of the plugin `rollup-plugin-sourcemap` which hasn't been updated for [4 years](https://github.com/maxdavidson/rollup-plugin-sourcemaps).

I opened a ticket 2 years ago to bring support for Rollup 3
- https://github.com/maxdavidson/rollup-plugin-sourcemaps/pull/126

Since rollup 4 has been released, and people have tried to make the maintainer give access, to allow someone to help maintain it or hand over the project:
- https://github.com/maxdavidson/rollup-plugin-sourcemaps/issues/130

I think it's about time we opt for a modernized fork, such as the one here, which support Rollup 4 and is a drop-in replacement:
- https://github.com/2wce/rollup-plugin-sourcemaps2

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 ### 🐞 Bug fixes
 - Fix circle won't render on mesa 24.1 with AMD GPU ([#4062](https://github.com/maplibre/maplibre-gl-js/issues/4062))
 - Fix hash router for urls ending with a hashtag ([#4730](https://github.com/maplibre/maplibre-gl-js/pull/4730))
+- Replace rollup-plugin-sourcemaps with rollup-plugin-sourcemaps2 ([#4740](https://github.com/maplibre/maplibre-gl-js/pull/4740))
 - _...Add new stuff here..._
 
 ## 4.7.0

--- a/package.json
+++ b/package.json
@@ -45,7 +45,6 @@
     "vt-pbf": "^3.1.3"
   },
   "devDependencies": {
-    "autoprefixer": "^10.4.20",
     "@mapbox/mapbox-gl-rtl-text": "^0.3.0",
     "@mapbox/mvt-fixtures": "^3.10.0",
     "@rollup/plugin-commonjs": "^26.0.1",
@@ -80,6 +79,7 @@
     "@typescript-eslint/eslint-plugin": "^7.18.0",
     "@typescript-eslint/parser": "^7.18.0",
     "address": "^2.0.3",
+    "autoprefixer": "^10.4.20",
     "benchmark": "^2.1.4",
     "canvas": "^2.11.2",
     "cssnano": "^7.0.6",
@@ -122,7 +122,7 @@
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
     "rollup": "^4.22.1",
-    "rollup-plugin-sourcemaps": "^0.6.3",
+    "rollup-plugin-sourcemaps2": "^0.4.1",
     "rw": "^1.3.3",
     "semver": "^7.6.3",
     "shuffle-seed": "^1.1.6",

--- a/rollup.config.ts
+++ b/rollup.config.ts
@@ -1,5 +1,5 @@
 import fs from 'fs';
-import sourcemaps from 'rollup-plugin-sourcemaps';
+import sourcemaps from 'rollup-plugin-sourcemaps2';
 import {plugins, watchStagingPlugin} from './build/rollup_plugins';
 import banner from './build/banner';
 import {RollupOptions} from 'rollup';

--- a/test/bench/rollup_config_benchmarks.ts
+++ b/test/bench/rollup_config_benchmarks.ts
@@ -1,5 +1,5 @@
 import fs from 'fs';
-import sourcemaps from 'rollup-plugin-sourcemaps';
+import sourcemaps from 'rollup-plugin-sourcemaps2';
 import replace from '@rollup/plugin-replace';
 import {plugins, nodeResolve} from '../../build/rollup_plugins';
 import commonjs from '@rollup/plugin-commonjs';

